File: lwjgl3/src/main/java/io/github/msameer0/rhythmicrush/lwjgl3/atlases/PackGamemodesAtlas.java
package io.github.msameer0.rhythmicrush.lwjgl3.atlases;

import com.badlogic.gdx.tools.texturepacker.TexturePacker;

public class PackGamemodesAtlas {
    public static void main(String[] args) {
        TexturePacker.Settings settings = new TexturePacker.Settings();
        settings.maxWidth = 2048;
        settings.maxHeight = 2048;
        settings.edgePadding = true;
        settings.duplicatePadding = true;

        TexturePacker.process(settings,
            "assets/game/objects/gamemodes", // input folder (one PNG per BlockType.textureName)
            "assets/game/objects",        // output folder
            "gamemodes"                      // atlas name → blocks.atlas + blocks.png
        );
    }
}

File: lwjgl3/src/main/kotlin/io/github/msameer0/rhythmicrush/lwjgl3/atlases/PackGamemodesAtlas.kt
package io.github.msameer0.rhythmicrush.lwjgl3.atlases

import com.badlogic.gdx.tools.texturepacker.TexturePacker

object PackGamemodesAtlas {
    @JvmStatic
    fun main(args: Array<String>) {
        val settings = TexturePacker.Settings()
        settings.maxWidth = 2048
        settings.maxHeight = 2048
        settings.edgePadding = true
        settings.duplicatePadding = true

        AtlasPackingUtils.packAllQualities(
            "textures_to_put_in_atlases/objects/gamemodes/cube",
            "assets/game/objects",
            "cubes",
            settings
        )

        AtlasPackingUtils.packAllQualities(
            "textures_to_put_in_atlases/objects/gamemodes/ship",
            "assets/game/objects",
            "ships",
            settings
        )
    }
}
